src/components/templates/Audience/Audience.tsx

```diff
@@ -174,13 +174,17 @@ export const Audience: React.FunctionComponent = () => {
   );
 
   useEffect(() => {
-    firebase
+    const unsubscribeListener = firebase
       .firestore()
       .collection("venues")
       .doc(venueId as string)
       .onSnapshot((doc) =>
         setIframeUrl(ConvertToEmbeddableUrl(doc.data()?.iframeUrl || "", true))
       );
+
+    return () => {
+      unsubscribeListener();
+    };
```

Comment:
We probably technically could do that, but I think it complicates the pattern. Normally the ‘cleanup function’ in a useEffect is specified in the format I’ve used here, then calls whatever cleanup needs to happen.

we could technically probably just return straight from the snapshot listener and I expect it would work, but it would be SUPER unobvious code then.